src/ripple/basics/tagged_integer.h

```diff
@@ -0,0 +1,223 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright 2014, Nikolaos D. Bougalis <nikb@bougalis.net>
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_UTILITY_TAGGED_INTEGER_H_INCLUDED
+#define BEAST_UTILITY_TAGGED_INTEGER_H_INCLUDED
+
+#include <ripple/beast/hash/hash_append.h>
+#include <boost/operators.hpp>
+#include <functional>
+#include <iostream>
+#include <type_traits>
+#include <utility>
+
+namespace ripple {
+
+/** A type-safe wrap around standard unsigned integral types
```

Comment:
The comment says this is a wrapper for `unsigned` types. However, there is nothing constraining it to unsigned. I'm assuming the comment is just out of date. If it _is_ meant to be for just unsigned, we should have a `static_assert`, and probably should not define unary `operator-`. If it's not meant to be constrained, we should just update the comment.